From 50cbc28c0ea83f5ad10eae2646a4da948bef8cbb Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Sat, 3 Aug 2019 15:00:49 +0200
Subject: [PATCH] CMakeLists.txt: respect library dirs

This, Gentoo specific patch respects the library names lib vs. lib64
for different platforms.

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 CMakeLists.txt | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62fe812..44d2407 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1094,14 +1094,14 @@ else ()
     file(WRITE "${pkgconfig_file}" "# draco pkg-config.\n")
     file(APPEND "${pkgconfig_file}" "prefix=${prefix}\n")
     file(APPEND "${pkgconfig_file}" "exec_prefix=${prefix}/bin\n")
-    file(APPEND "${pkgconfig_file}" "libdir=${prefix}/lib\n")
+    file(APPEND "${pkgconfig_file}" "libdir=${prefix}/lib${LIB_SUFFIX}\n")
     file(APPEND "${pkgconfig_file}" "includedir=${prefix}/include\n\n")
     file(APPEND "${pkgconfig_file}" "Name: Draco\n")
     file(APPEND "${pkgconfig_file}" "Description: 3D Compression.\n")
     file(APPEND "${pkgconfig_file}" "Version: ${draco_version}\n")
     file(APPEND "${pkgconfig_file}" "Requires:\n")
     file(APPEND "${pkgconfig_file}" "Conflicts:\n")
-    file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib -ldraco\n")
+    file(APPEND "${pkgconfig_file}" "Libs: -L${prefix}/lib${LIB_SUFFIX} -ldraco\n")
     file(APPEND "${pkgconfig_file}" "Cflags: -I${prefix}/include -std=c++11\n")
   endif ()
 
@@ -1134,19 +1134,19 @@ else ()
   if (NOT MSVC)
     # Install our pkg-config file.
     install(FILES "${draco_build_dir}/draco.pc"
-            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig")
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/pkgconfig")
   endif ()
 
   # Add install rules for lib and executable targets.
   install(TARGETS dracodec dracoenc draco
-          DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+          DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   if (BUILD_UNITY_PLUGIN)
     install(TARGETS dracodec_unity
-            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   endif ()
   if (BUILD_MAYA_PLUGIN)
     install(TARGETS draco_maya_wrapper
-            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
   endif ()
 
   install(TARGETS draco_decoder draco_encoder DESTINATION
@@ -1154,9 +1154,9 @@ else ()
 
   # Generate a CMake config file.
   set(draco_include_install_dir "include/draco")
-  set(draco_lib_install_dir "lib")
+  set(draco_lib_install_dir "lib${LIB_SUFFIX}")
   set(draco_script_install_dir
-      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/draco/cmake")
+      "${CMAKE_INSTALL_PREFIX}/${draco_lib_install_dir}/cmake/drako")
   set(draco_config_in_file "${draco_root}/cmake/DracoConfig.cmake")
   set(draco_config_out_file "${draco_build_dir}/DracoConfig.cmake")
 
@@ -1164,7 +1164,7 @@ else ()
       "${draco_config_in_file}"
       "${draco_config_out_file}"
       NO_CHECK_REQUIRED_COMPONENTS_MACRO
-      INSTALL_DESTINATION "${draco_lib_install_dir}/draco/cmake"
+      INSTALL_DESTINATION "${draco_lib_install_dir}/cmake/drako"
       PATH_VARS draco_include_install_dir draco_lib_install_dir)
   write_basic_package_version_file(
       "${draco_build_dir}/DracoConfigVersion.cmake"
-- 
2.22.0

